frontend/lib/resolution/resolution-error-classes-list.cpp

```diff
@@ -117,14 +109,39 @@ static void describeSymbolSource(ErrorWriterBase& wr,
       wr.note(locationOnly(elt.visibilityClauseId), errbegin, " the '", elt.visibilityStmtKind, "' statement", nameSuffix, " here:");
       wr.code<ID,ID>(elt.visibilityClauseId, { elt.visibilityClauseId });
       from = elt.renameFrom;
-      first = false;
+      needsIntroText = false;
     }
   }
+}
+
+// describe where a symbol came from
+// (in a way, it prints out a ResultVisibilityTrace)
+// 'start' indicates where in the trace to start, since sometimes
+// the first element might have already been printed.
+// 'oneOnly' indicates that only the 1st match should be described
+//
+// if 'intro' will be emitted before the first message for a trace
+// (only relevant if start==0). If it is not empty, it should probably
+// end with a space.
+static void describeSymbolSource(ErrorWriterBase& wr,
```

Comment:
I've been thinking it might be better to only have `describeSymbolTrace` anyway. I will look at that.

Anyway, the difference is whether they describe the symbol found itself, or just how it was found.